### ISSUE TITLE
Don't hesitate to create an issue if you have some question...

### ISSUE THREAD

**Issue by nennigb** [nennigb/amc2moodle#4]
Hi,

Feel free to ask your questions here :

    on the usage
    on bugs
    ...

In case of trouble during the conversion, put the command line output to help to locate the origin of the problem.

nennigb

**Comment by spinchinat** [comment]
I cannot run "pip install ."
ERROR: Directory '.' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.

and I have no idea where the setup.py file can be.

**Comment by spinchinat** [comment]
I have run  amc2moodle source.tex -o quizLOG.xml -c catname
and this is the end of the entire process. Can anyone help here?

Conversion complete: No obvious problems.
processing finished Mon Jun  8 14:28:12 2020
 > Running Python conversion...
bareme local : {'p': '0', ' formula': ' ((NB > 1) ? ((NBC-NMC)/NB) : NBC)'}
Traceback (most recent call last):
  File "/usr/local/bin/amc2moodle", line 107, in <module>
    run()
  File "/usr/local/bin/amc2moodle", line 98, in run
    indentXML=indentFlag, usetempdir=tempDir)
  File "/usr/local/lib/python3.7/site-packages/amc2moodle/amc2moodle/amc2moodle_class.py", line 142, in __init__
    self.runBuilding()
  File "/usr/local/lib/python3.7/site-packages/amc2moodle/amc2moodle/amc2moodle_class.py", line 227, in runBuilding
    deb=self.deb)
  File "/usr/local/lib/python3.7/site-packages/amc2moodle/amc2moodle/convert.py", line 391, in to_moodle
    if (float(amc_bml['b']) < 1):
KeyError: 'b'

**Comment by nennigb** [comment]
Hi,

I suspect a problem due to local scoring (bareme in french). We have implement scoring at quiz level (`\baremeDefautS{e=-0.5,b=1,m=-0.5}`) or at question level but not yet at answer level. In all case, you should use the complete scoring syntax (the only supported one)
```
\bareme{e=-0.5,b=1,m=-1.,p=-0.5} or \scoring{e=-0.5,b=1,m=-1.,p=-0.5}
```
you can have a look on this [exemple](https://github.com/nennigb/amc2moodle/blob/master/amc2moodle/amc2moodle/test/QCM.tex) for supported scoring syntaxes.

In a first time, I suggest to simplify your scoring strategy and just comment such command and run again `amc2moodle` to identify the problem.

Let me know if it fixes your problem. If not, could you attach a minimal exemple (only one question, no sensitive information) to reproduce the error ?  

Benoit

ps : Because moodle scoring is different from AMC, you also should set `b=1`.

**Comment by pablo-angulo** [comment]
Hello, Benoit!
I think you are not using the \explain{} command to give feedback to the students... is there a place in the [moodle xml](https://docs.moodle.org/310/en/Moodle_XML_format) for such explanations? It appears that it is possible to give feedback to each individual answer, not to the whole question.
Regards

**Comment by nennigb** [comment]
Hi,

You're right. I never plug the feedback between moodle and amc.
In moodle you can specified feedback for each answer using
```
<answer fraction="0" format="html">
      <text><![CDATA[<p>This is a question</p>]]></text>
      <feedback format="html">
        <text></text>
      </feedback>
  </answer>
```
or and at question level, using
```
<generalfeedback format="html">
      <text>always given</text>
</generalfeedback
<correctfeedback format="html">
      <text><![CDATA[<p>If everything is correct <br></p>]]></text>
</correctfeedback>
<partiallycorrectfeedback format="html">
      <text><![CDATA[<p>If some part are correct<br></p>]]></text>
</partiallycorrectfeedback>
<incorrectfeedback format="html">
      <text><![CDATA[<p>if everything is wrong<br></p>]]></text>
</incorrectfeedback>
```
In my opinion, mapping `\explain` content to general feedback, could make sense. What do you think ?

**Comment by pablo-angulo** [comment]
I agree that is the right way^{TM}

**Comment by nennigb** [comment]
I will do it in few days...

**Comment by nennigb** [comment]
It should work now. You can clone the master or get the new version (2.7) on pipy.
For all details on new possibilities see the PR #43 notes.

**Comment by pablo-angulo** [comment]
Perfect! The moodle xml looks fine and moodle got the idea! Thanks, Benoit

**Comment by pablo-angulo** [comment]
Suggestion: advertise in bigger letters that `amc2moodle` expands LaTeX commands, but alternatives like the `moodle` ctan package do not.

Actually, in the project [pyexams](https://framagit.org/pang/pyexams) I tried to use the `moodle` ctan project before `amc2moodle`, because `amc2moodle`, using `LaTeXML`, is slower, until I noticed this difference between them, which a very important feature for potential many users.

`amc2moodle` actually rocks!
